### DISCUSSION TITLE
Connecting a remote sensor

### DISCUSSION THREAD

**Discussion by Dattatech** [hassio-addons/app-zerotier#100] [Q&A]
Hi,

following scenario.
I have my home assistant installation at site number one, with it's own WIFI.

Then I have around 40 other sites, where I need to collect LUX values. I am using an ESP32 + BH1750 sensor at each site. Each site has it's own WIFI.

My question:
Can zerotier be used to allow the ESP32 + Sensor to send values to the home assistant installation, even if they are not on the same WIFI, or do I have to use a home assistant installation at each site and then connect each of the ESPs to their own HA installation?

**Comment by frenck** [comment]
In theory, that is possible. If you bridge/route both networks over your Zerotier network, they can reach each other. However, that is not a click-done solution.

**Reply by Dattatech** [reply]
ok, thanks for the reply Franck. I think that will be too much interference with the WIFIs of the 40 different sites. I suppose, the best way would be to give each site, it's own HA on RasPi Installation, let the local ESP32s talk to their own HA installation and then let the 40 installations communicate the data with the main HA installation. What do you think would be the best way?

Thanks,
Pradeep

**Reply by frenck** [reply]
I would personally not recommend connecting ESP devices remotely like you are suggesting. However, your question was about if it was possible, and sure it is.

I personally would use an MQTT broker when using multiple locations.

**Reply by Dattatech** [reply]
Very true, I did ask about that. In fact, I am also looking into the MQTT approach. So presently I am looking for the optimal and most secure way to connect the outer lighting of a chain of stores. Switching it on and off remotely as well as get feedback through LUX sensor if the lights stop working.

Your input has been most valuable Franck! Thanks a lot!
